OpenloginSwiftSdkDemo/OpenloginSwiftSdkDemo/ContentView.swift

```diff
@@ -1,47 +1,94 @@
 import SwiftUI
 import OpenLogin
 
+
 struct ContentView: View {
+    @SwiftUI.State var text = ""
     var body: some View {
-        Button(
-            action: {
-                OpenLogin
-                    .webAuth()
-                    .start {
-                        switch $0 {
-                        case .success(let result):
-                            print("""
-                                Signed in successfully!
-                                    Private key: \(result.privKey)
-                                    User info:
-                                        Name: \(result.userInfo.name)
-                                        Profile image: \(result.userInfo.profileImage ?? "N/A")
-                                        Type of login: \(result.userInfo.typeOfLogin)
-                                """)
-                        case .failure(let error):
-                            print("Error: \(error)")
+        VStack {
+            Button(
+                action: {
+                    OpenLogin
+                        .webAuth()
+                        .login(OLLoginParams()) {
+                            switch $0 {
+                            case .success(let result):
+                                    showResult(result: result)
+                            case .failure(let error):
+                                print("Error: \(error)")
+                            }
+                        }
+                },
+                label: {
+                    Text("Sign In")
+                        .padding()
+                }
+            )
+            
+            Button(
+                action: {
+                    OpenLogin
+                        .webAuth()
+                        .login(OLLoginParams(provider: .GOOGLE)) {
+                            switch $0 {
+                            case .success(let result):
+                                showResult(result: result)
+                            case .failure(let error):
+                                print("Error: \(error)")
+                            }
                         }
-                    }
-            },
-            label: {
-                Text("Sign In")
-                    .padding()
-            }
-        )
-        Button(
-            action: {
-                OpenLogin
-                    .webAuth()
-                    .signOut()
-                print("signed out.")
-            },
-            label: {
-                Text("Sign Out")
-                    .padding()
-            }
-        )
+                },
+                label: {
+                    Text("Sign In with Google")
+                        .padding()
+                }
+            )
+            
+            Button(
+                action: {
+                    OpenLogin
+                        .webAuth()
```

Comment:
why is the api like OpenLogin.webAuth ?

Reply:
I guess it is because the orignial implemener intened to have multiple login flow, like maybe login with ios native mechanisms without browsers? i dont think it is possible through.

Reply:
it's not possible. pls refactor this to maintain same api as android

Reply:
sure